update.py

```
@@ -116,20 +124,40 @@ def update_version_info(version: str):
 
+def install_dependencies():
+    """ Install the required dependencies for the script. """
+    for package in package_names:
+        # Skip installing if the module is already installed
+        if importlib.util.find_spec(package) is not None:
+            continue
+        else:
+            print(f"Package '{package}' not found - attempting to install...")
+
+        try:
+            subprocess.run(["pip", "install", package], check=True, capture_output=True)
```

Ruff (0.14.0)

`subprocess` call: check for execution of untrusted input

(S603)